File: enode/src/main/java/com/enodeframework/domain/IAggregateStorage.java
package com.enodeframework.domain;

import java.util.concurrent.CompletableFuture;

/**
 * Represents an aggregate storage interface.
 */
public interface IAggregateStorage {
    /**
     * Get an aggregate from aggregate storage.
     *
     * @param aggregateRootType
     * @param aggregateRootId
     * @param <T>
     * @return
     */
    <T extends IAggregateRoot> CompletableFuture<T> getAsync(Class<T> aggregateRootType, String aggregateRootId);
}

File: enode/src/main/java/org/enodeframework/domain/IAggregateStorage.kt
package org.enodeframework.domain

import java.util.concurrent.CompletableFuture

/**
 * Represents an aggregate storage interface.
 */
interface IAggregateStorage {
    /**
     * Get an aggregate from aggregate storage.
     */
    fun <T : IAggregateRoot?> getAsync(aggregateRootType: Class<T>, aggregateRootId: String): CompletableFuture<T>
}